src/jupyterscad/_openscad.py

```diff
@@ -0,0 +1,85 @@
+"""
+Jupyter SCAD
+Copyright (C) 2023 Jennifer Reiber Kyle
+
+This program is free software: you can redistribute it and/or modify it under
+the terms of the GNU General Public License as published by the Free Software
+Foundation, either version 3 of the License, or (at your option) any later
+version.
+
+This program is distributed in the hope that it will be useful, but WITHOUT ANY
+WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A
+PARTICULAR PURPOSE. See the GNU General Public License for more details.
+
+You should have received a copy of the GNU General Public License along with
+this program. If not, see <https://www.gnu.org/licenses/>.
+"""
+import logging
+import platform
+import subprocess
+from pathlib import Path
+from shutil import which
+
+from .exceptions import OpenSCADException
+
+LOGGER = logging.getLogger(__name__)
+
+DEFAULT_OPENSCAD_EXECUTABLE = {
+    "Linux": ["/usr/bin/openscad", "/usr/local/bin/openscad"],
+    "Darwin": ["/Applications/OpenSCAD.app/Contents/MacOS/OpenSCAD"],
+}
+
+
+def process(scad_file, output_file, executable: Path = None):
+    """Generate stl from scad using OpenSCAD executable"""
+    if executable and not Path(executable).is_file():
+        raise OpenSCADException(f"Specified executable {executable} does not exist.")
+
+    if not executable:
+        executable = detect_executable()
+
+    cmd = [executable, "-o", output_file, scad_file]
+    LOGGER.info(cmd)
+    try:
+        subprocess.run(cmd, check=True, capture_output=True, text=True)
+    except subprocess.CalledProcessError as e:
+        raise OpenSCADException(str(e.stderr))
+
+
+def detect_executable() -> Path:
```

Comment:
```python
def detect_executable2() -> Path:
    macPath = "/Applications/OpenSCAD.app/Contents/MacOS/OpenSCAD"
    executable = which("openscad") or \
                 which("OpenSCAD") or \
                 which(macPath)

    if not executable:
        raise OpenSCADException(
            f"OpenSCAD executable not found. Couldn't find a openscad or "
            "OpenSCAD executable in PATH nore could I find a executable at "
            "{macPath}. Please specify the path to the OpenSCAD executable"
            )

    LOGGER.debug(f"Executable path ({executable}) found.")
    return Path(executable)
```

I think this does exactly the same (the logging messages are a little different).

Reply:
oh yeah that's cleaner. when I have OpenSCAD added to my path, both `which('openscad')` and `which('OpenSCAD')` pick it up. Are you noticing it is capitalization-sensitive on your machine?

Reply:
make macPath a dictionary with system.platform as key for multi platform support.

Reply:
yep:

```python
>>> shutil.which("OpenSCAD")
>>> shutil.which("openscad")
'/home/jeff/.local/bin/openscad'

```

Reply:
gotcha. so I can see why we want `shutil.which("openscad")` but don't see the case for `shutil.which("OpenSCAD")`

Reply:
I though since the macpath was case sensitive it's supposed to be like that on macos....

I've no idea which executable names are or might be used....... it would be pure guessing....

Reply:
lol ok `shutil.which("openscad")` works for macos as well if it's in the path. So I'll just stick with that one and then the mac and windows directories.